dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ResourcePolicyRestRepository.java

```diff
@@ -52,9 +72,90 @@ public ResourcePolicyRest findOne(Context context, Integer id) {
         throw new RepositoryMethodNotImplementedException(ResourcePolicyRest.NAME, "findAll");
     }
 
-
     @Override
     public Class<ResourcePolicyRest> getDomainClass() {
         return ResourcePolicyRest.class;
     }
+
+    @PreAuthorize("hasAuthority('AUTHENTICATED')")
+    @SearchRestMethod(name = "resource")
+    public Page<ResourcePolicyRest> resource(@Parameter(value = "uuid", required = true) UUID resourceUuid,
+                                      @Parameter(value = "action", required = false) String action, Pageable pageable) {
+
+        List<ResourcePolicy> resourcePolisies = null;
+        int total = 0;
+        try {
+            Context context = obtainContext();
+            if (action != null) {
+                int actionId = Constants.getActionID(action);
+                resourcePolisies = resourcePolicyService.searchByResouceUuidAndActionId(context, resourceUuid, actionId,
+                        pageable.getOffset(), pageable.getOffset() + pageable.getPageSize());
+                total = resourcePolicyService.searchCountByResouceAndAction(context, resourceUuid, actionId);
+            } else {
+                resourcePolisies = resourcePolicyService.searchByResouceUuid(context, resourceUuid,
+                        pageable.getOffset(), pageable.getOffset() + pageable.getPageSize());
+                total = resourcePolicyService.searchCountByResourceUuid(context, resourceUuid);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        return converter.toRestPage(resourcePolisies, pageable, total, utils.obtainProjection(true));
+    }
+
+    @PreAuthorize("hasAuthority('AUTHENTICATED')")
```

Comment:
due to https://jira.lyrasis.org/browse/DS-4199 this will be ineffective. In any case at this stage (in this PR) your are not yet testing the security constraints stated in the contract so it is fine for me